packages/preferences/src/browser/views/preferences-widget-bindings.ts

```diff
@@ -0,0 +1,80 @@
+/********************************************************************************
+ * Copyright (C) 2020 Ericsson and others.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the Eclipse
+ * Public License v. 2.0 are satisfied: GNU General Public License, version 2
+ * with the GNU Classpath Exception which is available at
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ ********************************************************************************/
+import { interfaces } from 'inversify';
+import { PreferencesWidget } from './preferences-widget';
+import { WidgetFactory, createTreeContainer, TreeWidget, TreeProps, defaultTreeProps, TreeDecoratorService } from '@theia/core/lib/browser';
+import { SinglePreferenceDisplayFactory } from './components/single-preference-display-factory';
+import { PreferencesTreeWidget } from './preferences-tree-widget';
+import { PreferencesEditorWidget } from './preferences-editor-widget';
+import { PreferencesSearchbarWidget } from './preferences-searchbar-widget';
+import { PreferencesScopeTabBar } from './preferences-scope-tabbar-widget';
+import { PreferencesDecorator } from '../util/preferences-decorator';
+import { PreferencesDecoratorService } from '../util/preferences-decorator-service';
+import { SinglePreferenceWrapper } from './components/single-preference-wrapper';
+
+export function bindPreferencesWidgets(bind: interfaces.Bind): void {
+    bind(PreferencesWidget).toSelf();
+    bind(WidgetFactory).toDynamicValue(({ container }) => ({
+        id: PreferencesWidget.ID,
+        createWidget: () => container.get(PreferencesWidget)
+    })).inSingletonScope();
+
+    bind(SinglePreferenceWrapper).toSelf();
+
+    bind(PreferencesTreeWidget).toDynamicValue(ctx =>
+        createPreferencesTree(ctx.container));
+    bind(WidgetFactory).toDynamicValue(context => ({
+        id: PreferencesTreeWidget.ID,
+        createWidget: (): PreferencesTreeWidget => createPreferencesTree(context.container),
+    })).inSingletonScope();
```

Comment:
You are calling `createPreferencesTree` twice here, maybe why you see two instances for `PreferencesTreeWidget`.

It looks like `createPreferencesTree` creates a new sub-container on each call, which means an isolated singleton cache?

I don't know if the following code is what you want to do, but maybe it will help:

```ts
    bind(PreferencesTreeWidget).toDynamicValue(ctx =>
        createPreferencesTree(ctx.container)
    ).inSingletonScope(); // ?
    bind(WidgetFactory).toDynamicValue(context => ({
        id: PreferencesTreeWidget.ID,
        createWidget: (): PreferencesTreeWidget => context.container.get(PreferencesTreeWidget),
    })).inSingletonScope();
```

Reply:
@marechal-p, Thanks! That snippet seems to have done the trick.